### ISSUE TITLE
[Razer Phone 2] [Android 10 GSI] Bluetooth audio not working

### ISSUE THREAD

**Issue by Centra-Plex** [phhusson/treble_experimentations#1131]
I tried various "fixes" to try to get Bluetooth audio up and running and haven't had any luck. I provided the log  for what happens when I tried connecting to a Bluetooth speaker to the device. 

[logcat.txt](https://github.com/phhusson/treble_experimentations/files/4242400/logcat.txt)

**Comment by xxzeroz** [comment]
Have you tried this method?

In TWRP
- Mount Vendor
- Go to file manager go to /vendor/overlay
- Delete BluetoothResCommon.apk  and  RazerAuraBluetoothRes.apk 
- Profit!

**Comment by Centra-Plex** [comment]
@xxzeroz Just tried Bluetooth won't even turn on now...

**Comment by shmilyhua** [comment]
This method is from Sticks210 in the post(https://forum.xda-developers.com/razer-phone-2/how-to/fix-bluetooth-audio-pie-gsi-t3918763/page4).

> To fix music on GSI (not aptx or anything, , just functionality), delete the BluetoothResCommon.apk, make a duplicate copy of RazerAuraBluetoothRes.apk, and rename it BluetoothResCommon.apk just like the one you deleted. Working for me in Pie AOSiP.

I've tried it on PE 10, Havoc 3.2 and Evo X 4.1, at least music does work now.

**Comment by Centra-Plex** [comment]
@shmilyhua thank you and actually I found another fix on the razer phone telegram that for some reason works. Before first boot delete Bluetooth res and razer Bluetooth res. It just works only real issue is bt calling not functional but I can live with that tbh.

**Comment by phhusson** [comment]
@Centra-Plex I'm not totally clear what you said is your fix.

You deleted RazerAuraBluetoothRes.apk, and BluetoothResCommon.apk and that's all?

**Comment by Centra-Plex** [comment]
@phhusson yeah for some reason before first boot if you delete those two files it seems to work. Only other issue I am occurring now is intermittent sms

**Comment by pietro-v** [comment]
To clarify, in call audio over Bluetooth does not function even after deleting BluetoothResCommon.apk and renaming RazerAuraBluetoothRes.apk to BluetoothResCommon.apk prior to first boot. This includes v213.

**Comment by RobotRoss** [comment]
Can also confirm this problem is present on the Razer Phone 1.
Deleting BluetoothResCommon.apk fixes Bluetooth Media audio, but in call audio remains broken. Using v221

**Comment by Centra-Plex** [comment]
> Can also confirm this problem is present on the Razer Phone 1.
> Deleting BluetoothResCommon.apk fixes Bluetooth Media audio, but in call audio remains broken. Using v221

No longer using the Razer Phone 2 but we found a fix on the telegram awhile ago. Here is how to preform the fix:

1) Wipe all your data and reinstall your GSI of Choice

2) Delete Bluetooth RES AND Aura RES APKs from the vendor file BEFORE first boot via TWRP

3) Boot into system

Done!

Also just curious RP1 has some official ROM support why are dealing with issues like this when you have official ROMs to choose from?

**Comment by RobotRoss** [comment]
> > Can also confirm this problem is present on the Razer Phone 1.
> > Deleting BluetoothResCommon.apk fixes Bluetooth Media audio, but in call audio remains broken. Using v221
> 
> No longer using the Razer Phone 2 but we found a fix on the telegram awhile ago. Here is how to preform the fix:
> 
> 1. Wipe all your data and reinstall your GSI of Choice
> 2. Delete Bluetooth RES AND Aura RES APKs from the vendor file BEFORE first boot via TWRP
> 3. Boot into system
> 
> Done!
> 
> Also just curious RP1 has some official ROM support why are dealing with issues like this when you have official ROMs to choose from?

Had one too many problems with Lineage which is the only ROM in development right now (any other active ROMs are based off of Lineage, so share the same bugs)

Anyway, unfortunately the Razer Phone 1 lacks the RazerAuraBluetoothRes.apk from the stock vendor image, so the problem is elsewhere

**Comment by pietro-v** [comment]
> > Can also confirm this problem is present on the Razer Phone 1.
> > Deleting BluetoothResCommon.apk fixes Bluetooth Media audio, but in call audio remains broken. Using v221
> 
> No longer using the Razer Phone 2 but we found a fix on the telegram awhile ago. Here is how to preform the fix:
> 
> 1. Wipe all your data and reinstall your GSI of Choice
> 2. Delete Bluetooth RES AND Aura RES APKs from the vendor file BEFORE first boot via TWRP
> 3. Boot into system
> 
> Done!
> 
> Also just curious RP1 has some official ROM support why are dealing with issues like this when you have official ROMs to choose from?

Thank you for the suggestion but in call Bluetooth audio does not work. This fixes media audio only.

**Comment by Centra-Plex** [comment]
> > > Can also confirm this problem is present on the Razer Phone 1.
> > > Deleting BluetoothResCommon.apk fixes Bluetooth Media audio, but in call audio remains broken. Using v221
> > 
> > No longer using the Razer Phone 2 but we found a fix on the telegram awhile ago. Here is how to preform the fix:
> > 
> > 1. Wipe all your data and reinstall your GSI of Choice
> > 2. Delete Bluetooth RES AND Aura RES APKs from the vendor file BEFORE first boot via TWRP
> > 3. Boot into system
> > 
> > Done!
> > 
> > Also just curious RP1 has some official ROM support why are dealing with issues like this when you have official ROMs to choose from?
> 
> Thank you for the suggestion but in call Bluetooth audio does not work. This fixes media audio only.

Huh odd it works on RP2.
